packages/flutter_tools/lib/src/run.dart

```diff
@@ -288,9 +293,78 @@ class RunAndStayResident {
     });
   }
 
+  DevFS devFS;
+
+  Future<Null> _updateDevFS() async {
+    if (devFS == null) {
+      devFS = new DevFS(Directory.current, observatory);
+
+      try {
+        await devFS.init();
+      } catch (error) {
+        devFS = null;
+        printError('Error initing DevFS: $error');
+        return null;
+      }
+    }
+
+    // Send the root and lib directories.
+    Directory directory = Directory.current;
+    _sendFiles(directory, '', _dartFiles(directory.listSync()));
+
+    directory = new Directory('lib');
+    _sendFiles(directory, 'lib', _dartFiles(directory.listSync(recursive: true, followLinks: false)));
```

Comment:
Hopefully none in here - `lib/`. I'm a bit paranoid about following symlinks because of the liberal use of them in dart projects. I'd like to try w/o it and course correct if we hit issues.